### PR TITLE
Browse d/copyright files

### PR DESCRIPTION
**Pull request by oorestisime**
Using the nav code from PR#4 this PR implements the browsing for d/copyright files.
Included in this branch:
- index view for the /copyright
- browsing by prefix
- browsing by list
- view versions for package
- api functions
- 404, 403, 500 error pages based on the sources BP

### CODE REVIEW

**Comment by matthieucan** [comment]
Thanks! I have 3 remarks for this PR:
- Templates are copied from the sources BP. Any way to avoid that?
- I think it's time to split test_webapp into test_web_sources and test_web_copyright. No hurry, this can be done later anyway.
- The search menu (on the top) makes the user exit /copyright, maybe we should discard it for this first version.

For the rest: well done, everything works fine! :)

**Comment by oorestisime** [comment]
Thanks for the review :)
- For the test suite i agree. i ll add this to the kanban board. 
- Concerning the search menu; it is included in templates/base.html. Should i create another general base for this case? Generally i did not create the base for the copyright BP since it is not going to be used atm (s.d.n/copyright uses the base of s.d.n and c.d.n is not deployed yet)
- For the templates i guess you are right. I ll move most of the templates (all besides index, license and maybe package) in the app/templates so that all the BP can use them.

**Comment by matthieucan** [comment]
On 31 May 2015 at 14:34, Orestis Ioannou notifications@github.com wrote:

> For the test suite i agree. i ll add this to the kanban board.

Thanks!

> Concerning the search menu; it is included in templates/base.html. Should i create another general base for this case? Generally i did not create the base for the copyright BP since it is not going to be used atm (s.d.n/copyright uses the base of s.d.n and c.d.n is not deployed yet)

Or maybe an option, something that is included or not... I'll let you
decide what's the best solution :)

> For the templates i guess you are right. I ll move most of the templates (all besides index, license and maybe package) in the app/templates so that all the BP can use them.

Perfect!

**Comment by oorestisime** [comment]
so i moved most of the templates in the core app templates. I think the errorhandler can be made BP agnostic since the bp_name is not used anymore. should i do that ?

**Comment by matthieucan** [comment]
Yes, even better!

File: debsources/tests/test_webapp.py [hunk 1]
```diff
@@ -199,6 +226,21 @@ def test_by_prefix(self):
...
 
+        # copyright BP
+        rv = self.app.get('/copyright/prefix/libc/')
+        self.assertIn("/cp/libcaca", rv.data)
+        # suite specified
+        rv = self.app.get('/copyright/prefix/libc/?suite=squeeze')
+        self.assertIn("/cp/libcaca", rv.data)
+        # a non-existing suite specified
+        rv = self.app.get(
+            '/copyright/prefix/libc/?suite=non-exsiting')
```
Comment (matthieucan) [inline debsources/tests/test_webapp.py]: typo here: exsiting

**Pull request merged**